classes/class-settings.php

```diff
@@ -1061,4 +1064,21 @@ public function get_settings_translations( $labels ) {
 
 		return $labels;
 	}
+
+	/**
+	 * Update actions dropdown options based on the connector selected.
+	 */
+	function update_actions() {
+		$connector = sanitize_text_field( $_POST['connector'] );
+		$actions = $GLOBALS['wp_stream']->connectors->term_labels['stream_action'];
+		if ( ! empty( $connector ) ) {
+			$class_name = '\WP_Stream\Connector_'.ucfirst( $connector );
+			if ( class_exists( $class_name ) ) {
+				$connector_class = new $class_name();
+				$actions         = $connector_class->get_action_labels();
+			}
+		}
+		ksort( $actions );
+		die( wp_json_encode( $actions ) );
```

Comment:
Can we use `wp_send_json_success()` instead here, and drop the `die()`?

The js would obviously need to be updated to support this, too.